chart/logging/providers/elasticsearch.vue

```diff
@@ -33,37 +34,8 @@ export default {
   },
 
   methods: {
-    fileUpload(field) {
-      this.toUpload = field;
-      this.$refs.uploader.click();
-    },
-
-    fileChange(event) {
-      const input = event.target;
-      const handles = input.files;
-      const names = [];
-
-      if ( handles ) {
-        for ( let i = 0 ; i < handles.length ; i++ ) {
-          const reader = new FileReader();
-
-          reader.onload = (loaded) => {
-            const value = loaded.target.result;
-
-            this.values[this.toUpload] = value;
-          };
-
-          reader.onerror = (err) => {
-            this.$dispatch('growl/fromError', { title: 'Error reading file', err }, { root: true });
-          };
-
-          names[i] = handles[i].name;
-          reader.readAsText(handles[i]);
-        }
-
-        input.value = '';
-      }
-    },
+    onCertSelected: createOnSelected('values.client_cert'),
+    onKeySelected:  createOnSelected('values.client_key')
```

Comment:
Why not emit the `input` event and use `v-model` with FileSelector instead of importing a special function to update values?

Reply:
I only think we should be using `v-model` if we have bi-directional dataflow and in this case I don't think the consumer of FileSelector should ever need to pass `:value` to FileSelector. Because I think this should be unidirectional I think it's better introduce a new event rather than to use `input` which has the bi-directional assumptions baked in.

If everyone would prefer that I use v-model to avoid the onSelected methods let me know and I'll change it.

Reply:
I agree with the event, since it's a "pick a file and something will happen" component (like a row in a table shows up) rather than an analog of `<input type="file">` that has a single stored value.  And it has to handle multiple somehow, which is obvious with events.

Reply:
If there aren't any other changes would someone mind merging?

Reply:
I thought you were going to replace the other usages with this

Reply:
Yeah, my original intent was to do it in a separate PR with the thought that a marge indicated it was okay. 

I went ahead and added the replacements in a new commit.